### PR TITLE
[Performance] Accelerate `_split_and_pad_sequence`

### PR DESCRIPTION
**Pull request by Blonck**
## Description

Adds a slightly simpler and faster version of `_split_and_pad_sequence`.

## Motivation and Context

While working on #1141  the inverse function `_inv_pad_sequence` was refactored and its performance increased. The same idea which was used in `_inv_pad_sequence` is used here to create the index for the mask.

On this way, unit tests for `_split_and_pad_sequence`, `_inv_pad_sequence, and `_get_num_per_traj` are created.

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).

### CODE REVIEW

**Review by vmoens — APPROVED** [review]
LGTM just a few edits
I would leave the linting do its job
there are errors on tests on gpu for tensordict, arange and splits are not on the same device.

File: benchmarks/test_objectives_utils.py
[the remark this answers is not in the document]
Comment (Blonck) [reply benchmarks/test_objectives_utils.py:1]: Or I just delete it. This benchmark was more for development to ensure that the new version is faster. Keeping it does not make sense for the long run, or?

File: test/test_objectives_value_utils.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+from torchrl.objectives.value.functional import _get_num_per_traj_init
+from torchrl.objectives.value.utils import (
+    _get_num_per_traj,
+    _inv_pad_sequence,
+    _split_and_pad_sequence,
+)
+
+
+class TestUtils:
+    @pytest.mark.parametrize("B", [None, (1, ), (4, ), (2, 2, ), (1, 2, 8, )])  # fmt: skip
```
[the remark this answers is not in the document]
Comment (Blonck) [reply test/test_objectives_value_utils.py:25]: It skips the automatic formatting from pre-commit. The formatter put each tuple of the list onto a new line.

File: test/test_objectives_value_utils.py
[the remark this answers is not in the document]
Comment (Blonck) [reply test/test_objectives_value_utils.py:1]: Should I move the test to test_costs.py?

File: torchrl/objectives/value/utils.py [hunk 2]
```diff
@@ -215,8 +217,10 @@ def _get_num_per_traj(dones_and_truncated):
     return num_per_traj
 
 
-def _split_and_pad_sequence(tensor, splits):
-    """Given a tensor of size [B, T, *other] and the corresponding traj lengths (flattened), returns the padded trajectories [NPad, Tmax, *other].
+def _split_and_pad_sequence(
+    tensor: Union[torch.Tensor, TensorDictBase], splits: torch.Tensor
+):
+    """Given a tensor of size [*B, T, F] and the corresponding traj lengths (flattened), returns the padded trajectories [NPad, Tmax, *other].
```
Comment (vmoens) [inline torchrl/objectives/value/utils.py:223]: if *other equal to F?

Comment (Blonck) [reply torchrl/objectives/value/utils.py:223]: Yes. The new version behaves as the previous one on all input shapes, or at least on all input shapes I've tested.
I just wanted to streamline the docu with other functions in functional.py.

Moreover, in practice all methods using `_split_and_pad_sequence` must also use `_get_num_per_traj` which needs input in the form of [*B, T]. Thus in all methods the input is anyway reshaped (`vec_generalized_advantage_estimate`, `reward2go`).

**Pull request merged**